delivery/services/staging_service.py

```diff
@@ -194,6 +203,61 @@ def stage_runfolder(self, runfolder_id, projects_to_stage=None, callback=None):
 
         return project_and_stage_order_ids
 
+    def _create_links_area_for_project_runfolders(self, project_name, projects):
+        """
+        Creates a directory in which it creates links to all runfolders for the projects
+        given. This is useful so that we can then rsync that directory to
+        the staging area.
+        :param project_name: name of the project
+        :param projects: runfolders with the specified project on them
+        :return: the path to the dir created
+        """
+
+        project_dir = os.path.join(self.project_links_directory, project_name)
+        try:
+            FileSystemService.mkdir(project_dir)
+        except FileExistsError as e:
+            log.warning("Project dir: {} already exists".format(project_dir))
+
+        for project in projects:
+            try:
+                link_name = os.path.join(project_dir, project.runfolder_name)
+                FileSystemService.symlink(project.path, link_name)
+            except FileExistsError:
+                log.warning("Project link: {} already exists".format(project_dir))
+                continue
+
+        return FileSystemService.abspath(project_dir)
+
+    def stage_runfolders_for_project(self, project_id, delivery_type):
+        projects = list(self.runfolder_project_repo.get_project(project_name=project_id))
+
+        if len(projects) < 1:
+            raise ProjectNotFoundException("Could not find any Project folders"
+                                           " for project name: {}".format(project_id))
+
+        # TODO
```

Comment:
I presume these will be defined somewhere, as I noticed you were referencing these in `StagingProjectRunfoldersHandler` as well.